src/vdom/index.js

```diff
@@ -1,5 +1,18 @@
 import { extend } from '../util';
+import options from '../options';
 
+/**
+ * Checks that two components are equal
+ * @param {Component} component1 First component to compare
+ * @param {Component} component2 Second component to compare
+ * @return {boolean}
+ * @private
+ */
+export function areComponentsEqual(component1, component2) {
+	return options.areComponentsEqual
```

Comment:
One could remove the `if` check here by making it the default implementation for `options.areComponentsEqual`

Reply:
Pro - speed matters
Cons - you will be unable to revert configuration. Not a big deal, but could be nasty.

Reply:
Probably one `if` here is way faster than one more function call you are asking for.

Reply:
@marvinhagemeister - you were right.
My variant is 4x times slower than yours, and yours are equal to the base line.

https://jsperf.com/call-or-if/1